src/Dfc.CourseDirectory.WebV2/BinaryStorageProvider/BlobStorageBinaryStorageProvider.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Auth;
+using Microsoft.Azure.Storage.Blob;
+
+namespace Dfc.CourseDirectory.WebV2.BinaryStorageProvider
+{
+    public class BlobStorageBinaryStorageProvider : IBinaryStorageProvider
+    {
+        private readonly CloudBlobClient _blobClient;
+        private readonly string _containerName;
+
+        public BlobStorageBinaryStorageProvider(
+            string accountName,
+            string accountKey,
+            string containerName)
+        {
+            var account = new CloudStorageAccount(new StorageCredentials(accountName, accountKey), useHttps: true);
+            _blobClient = account.CreateCloudBlobClient();
+
+            _containerName = containerName;
+        }
+
+        public async Task<bool> TryDownloadFile(string path, Stream destination)
+        {
+            var blobRef = _blobClient.GetContainerReference(_containerName).GetBlockBlobReference(path);
+
+            if (!await blobRef.ExistsAsync())
+            {
+                return false;
+            }
+
+            await blobRef.DownloadToStreamAsync(destination);
```

Comment:
Worth perhaps checking the result to be safe on the potential race condition?

Reply:
What's the race?

Reply:
Blob could be deleted in between `Exists()` check and `Download()`.

Suggest `try { ... } catch (StorageException ex) when (ex.StatusCode == 412) { return false; }` (Don't quote me on the exception and code type.)

Reply:
Nothing ever deletes blobs here

Reply:
That's no guarantee that a different API/consumer of this data won't in the future.